common/pubsub/pubsub_sync.go

```diff
@@ -0,0 +1,43 @@
+package pubsub
+
+import (
+	"context"
+
+	"github.com/status-im/status-go/common"
+)
+
+// PublishSync will send the value val of type T on the specified Publisher. Subscribers do not block each other
+// but the call will block until the event is processed by all subscribers.
+// If the context is canceled, the value may not be sent to all subscribers.
+func PublishSync[T any](ctx context.Context, p *Publisher, val T) {
+	subs := getTypePublisher[T](p)
+	subs.Publish(ctx, val)
+}
+
+// SubscribeSync creates a channel to listen for events of type T published by the Publisher.
+// The returned channel will be closed by the Publisher.
+// UnsubFn must be called by the subscriber before it becomes unable to process events. It will
+// block until the Publisher is closed, ensuring all events are processed.
```

Comment:
That's a bit unexpected. This makes subscribers tightly coupled to publisher lifetime. For instance, if there is one global publisher in the app, subscribers will only accumulate over time.